src/main/java/net/helix/pendulum/crypto/Merkle.java

```diff
@@ -70,18 +113,11 @@
             // Take two following keys (i=0: (k0,k1), i=1: (k2,k3), ...) and get one crypto of them
```

Comment:
Shouldn't the tree depth be equal to the `log  base 2` of the `leaves.size()` rather than `Math.sqrt`? Here are some efficient implementations: https://stackoverflow.com/questions/3305059/how-do-you-calculate-log-base-2-in-java-for-integers

Reply:
Indeed log base 2 should be used not sqrt even if for small numbers both functions will return the same result (except 1)

Reply:
Fix with: (int) Math.ceil((float)(leavesNumber / Math.log(2))); (adding extra epsilon didn't fix the problem, so I have casted to float and the less signifiant bits from precision are lost)